plugins/PrivacyManager/Tracker/RequestProcessor.php

```diff
@@ -37,15 +38,24 @@ public function manipulateRequest(Request $request)
         }
     }
 
+    /**
+     * pseudo anonymization as we need to make sure to always generate the same UserId for the same original UserID
+     *
+     * @param $userId
+     * @return string
+     */
     public static function anonymizeUserId($userId)
     {
-        // pseudo anonymization as we need to make sure to always generate the same UserId for the same original UserID
-        return sha1($userId . SettingsPiwik::getSalt());
+        $trackerCache = Tracker\Cache::getCacheGeneral();
+        $salt = '';
+        if (!empty($trackerCache[PrivacyManager::OPTION_USERID_SALT])) {
+            $salt = $trackerCache[PrivacyManager::OPTION_USERID_SALT];
+        }
+        return sha1($userId . $salt);
```

Comment:
If no salt is set, I think it should return the raw `$userId`. Otherwise you eventually create two different userIds for the same userId. It should be only for a short amount of time though I presume so may be fine to not use return raw `$userId`. However, it makes it also easier to discover when there is a bug as currently we would not notice if `$trackerCache[PrivacyManager::OPTION_USERID_SALT])` is never set.